### ISSUE TITLE
linter: `vitest/require-hook` false positive on vue/jsx files

### ISSUE THREAD

**Issue by Perdolique** [oxc-project/oxc#22160]
### What version of Oxlint are you using?

1.63.0

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "plugins": [
    "vitest"
  ],

  "categories": {
    "correctness": "off",
    "nursery": "off",
    "pedantic": "off",
    "perf": "off",
    "restriction": "off",
    "style": "off",
    "suspicious": "off"
  },

  "rules": {
    "vitest/require-hook": "error"
  }
}
```

### What happened?

Rule `vitest/require-hook` reports errors on Vue (Nuxt) / JSX files. Other `vitest/*` rules work correctly.

```text
  × vitest(require-hook): Require setup and teardown code to be within a hook.
    ╭─[app/pages/account.vue:51:3]
 50 │     
 51 │ ╭─▶   definePageMeta({
 52 │ │       layout: 'page'
 53 │ ╰─▶   })
 54 │     
    ╰────
  help: This should be done within a hook
```

```text
  × vitest(require-hook): Require setup and teardown code to be within a hook.
     ╭─[new-design-assets/scripts/view-detail.jsx:135:1]
 134 │ 
 135 │ Object.assign(window, { ItemDetailView });
     · ─────────────────────────────────────────
     ╰────
  help: This should be done within a hook
```

**Comment by kapobajza** [comment]
Hello @Perdolique. There's a problem with your config. Plugins like `vitest`, `jest` and similar, should only be run on test files. Even in `eslint-plugin-vitest` you have to [set up the config](https://github.com/vitest-dev/eslint-plugin-vitest/tree/main#usage) to only include test files. You should update your config to something like this:

_.oxlintrc.json_
```diff
{
-  "plugins": [
-    "vitest"
-  ],

  "categories": {
    "correctness": "off",
    "nursery": "off",
    "pedantic": "off",
    "perf": "off",
    "restriction": "off",
    "style": "off",
    "suspicious": "off"
  },

+  "overrides": [
+    {
+      "files": ["*.test.[t|j]s", "*.spec.[t|j]s"],
+      "plugins": ["vitest"],
+      "rules": {
+        "vitest/require-hook": "error",
+      },
+    },
+  ],

-  "rules": {
-    "vitest/require-hook": "error"
-  }
}
```

**Comment by Perdolique** [comment]
@kapobajza I have all other Vitest rules enabled as well, but only this one is reporting other files. Should at least one other Vitest rule have the same issue in that case?

**Comment by kapobajza** [comment]
> [@kapobajza](https://github.com/kapobajza) I have all other Vitest rules enabled as well, but only this one is reporting other files. Should at least one other Vitest rule have the same issue in that case?

That's because most, if not all, other rules are related to test specific stuff (like the `expect`, `test`, `afterEach`, `beforeEach` functions and similar). `vitest/require-hook` is also related to test specific stuff, but if it detects expressions at the top of the file, it will be triggered regardless of the context.

As an example, I added this global rule to my _.oxlintrc.json_ file:

```json
"rules": {
  "vitest/no-disabled-tests": "error"
}
```

And then I added a `test.skip` call in my _App.vue_ file:

```vue
<script setup lang="ts">
import { test } from 'vitest'

test.skip('something', () => {})
</script>
```

And when I ran `oxlint`, I got this error:

```
  × vitest(no-disabled-tests): Disabled test
   ╭─[src/App.vue:4:1]
 3 │ 
 4 │ test.skip('something', () => {})
   · ─────────
 5 │ </script>
   ╰────
  help: Remove the appending `.skip`

Found 0 warnings and 1 error.
Finished in 23ms on 1 file with 2 rules using 8 threads.
```

**Comment by Perdolique** [comment]
Alright. Seems like an inconvenience in case you need to extend specific shared config for a specific file pattern, as it's not supported. But okay, seems legit overall.
